nltk/tag/crfsuite.py

```diff
@@ -0,0 +1,162 @@
+    
+    >>> from nltk.tag.crfsuite import CRFTagger
+    >>> ct = CRFTagger()
+ 
+    >>> train_data = [[('University','Noun'), ('is','Verb'), ('a','Det'), ('good','Adj'), ('place','Noun')],
+    ... [('dog','Noun'),('eat','Verb'),('meat','Noun')]]
+    
+    >>> ct.train(train_data,'model.crf.tagger')
+    >>> ct.tag_sents([['dog','is','good'], ['Cat','eat','meat']])
+    [[('dog', 'Noun'), ('is', 'Verb'), ('good', 'Adj')], [('Cat', 'Noun'), ('eat', 'Verb'), ('meat', 'Noun')]]
+    
+    >>> gold_sentences = [[('dog','Noun'),('is','Verb'),('good','Adj')] , [('Cat','Noun'),('eat','Verb'), ('meat','Noun')]] 
+    >>> ct.evaluate(gold_sentences) 
+    1.0
+    
+    Setting learned model file  
+    >>> ct = CRFTagger() 
+    >>> ct.set_model_file('model.crf.tagger')
+    >>> ct.evaluate(gold_sentences)
+    1.0
+    
+    """
+    
+    def __init__(self, file_path=''):
+               
+        self._model_file = ''
+    
+
+    def set_model_file(self, model_file):
+        self._model_file = model_file
+            
+    def _get_features(self,data):
+        """
+        Extract basic features about this word including 
+             - Current Word 
+             - Is Capitalized ?
+             - Has Punctuation ?
+             - Has Number ?
+             - Suffixes up to length 3 
+        :return : a string which contains the features   
+        
+        """ 
+        feature_list = []  
+        # Capitalization 
+        if data[0].isupper():
+            feature_list.append('CAPITALIZATION')
+        
+        # Number 
+        pattern = re.compile('\\d')
+        if re.search(pattern, data) is not None:
+            feature_list.append('HAS_NUM') 
+        
+        # Punctuation
+        punc_cat = set(["Pc", "Pd", "Ps", "Pe", "Pi", "Pf", "Po"])
+        if all (unicodedata.category(x) in punc_cat for x in data):
+            feature_list.append('PUNCTUATION')
+        
+        # Suffix up to length 3
+        if len(data) > 1:
+            feature_list.append('SUF_' + data[-1:]) 
+        if len(data) > 2: 
+            feature_list.append('SUF_' + data[-2:])    
+        if len(data) > 3: 
+            feature_list.append('SUF_' + data[-3:])
+            
+        feature_list.append('WORD_' + data )
+        
+        return feature_list
+        
+    def tag_sents(self, sents):
+        '''
+        Tag a list of sentences. NB before using this function, user should specify the mode_file either by 
+                       - Train a new model using ``train'' function 
+                       - Use the pre-trained model which is set via ``set_model_file'' function  
+        :params sentences : list of sentences needed to tag. 
+        :type sentences : list(list(str))
+        :return : list of tagged sentences. 
+        :rtype : list (list (tuple(str,str))) 
+        '''
+        tagger = pycrfsuite.Tagger()
+        tagger.open(self._model_file)
```

Comment:
The parser shouldn't load and realod the tagger each time tag_sents is called. It makes more sense to do it in a constructor.